### PR TITLE
feat: asset head command

### PR DESCRIPTION
**Pull request by theus77**
|Q              |A  |
|---------------|---|
|Bug fix?       |n|
|New feature?   |y|	
|BC breaks?     |n|	
|Deprecations?  |n|	
|Fixed tickets  |n|	

New ems:asset:head command to update asset's seen information

### CODE REVIEW

File: src/Command/Asset/HeadAssetCommand.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        $this->fileService = $fileService;
+        $this->logger = $logger;
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setName('ems:asset:head')
```
Comment (dameert) [inline src/Command/Asset/HeadAssetCommand.php:31]: Fyi name is configured now via a private property: https://symfony.com/doc/current/console.html

protected static $defaultName = 'ems:asset:head';

**Comment by Davidmattei** [comment]
https://github.com/ems-project/EMSCoreBundle/issues/649

File: src/Command/Asset/HeadAssetCommand.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                if (!$assetUpload instanceof UploadedAsset) {
+                    throw new \RuntimeException('Unexpected UploadedAsset type');
+                }
+                $headIn = $this->fileService->headIn($assetUpload);
+                if (\count($headIn) > 0) {
+                    ++$found;
+                } else {
+                    ++$notFound;
+                }
+                ++$from;
```
Comment (dameert) [inline src/Command/Asset/HeadAssetCommand.php:58]: I think $notFound can be calculated, as ++$from will give you the exact total, thus $from - $found = $found

File: src/Command/Asset/HeadAssetCommand.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            }
+        }
+        $this->io->progressFinish();
+        if (0 !== $notFound) {
+            $this->io->warning(\sprintf('%d assets have not been found from %d', $notFound, $counter));
+        } else {
+            $this->io->success(\sprintf('%d assets have been found', $counter));
+        }
+
+        return 0;
```
Comment (dameert) [inline src/Command/Asset/HeadAssetCommand.php:69]:
```suggestion
        return Command::SUCCESS;
```

File: src/Entity/UploadedAsset.php [hunk 4]
```diff
@@ -101,6 +101,23 @@ class UploadedAsset implements EntityInterface
      */
     private $hashAlgo;
 
+    /**
+     * @ORM\Column(name="hidden", type="boolean", options={"default" : 0})
+     */
+    private bool $hidden;
```
[the remark this answers is not in the document]
Comment (theus77) [reply src/Entity/UploadedAsset.php:107]: It's for futur needs: hide an asset from the CK editor

File: src/Resources/DoctrineMigrations/pdo_pgsql/Version20210412122119.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
+<?php
```
Comment (dameert) [inline src/Resources/DoctrineMigrations/pdo_pgsql/Version20210412122119.php:1]: This reminds me that I wanted to talk about abandonning mysql and sqlite DoctrineMigrations as they are not systematically maintained.

File: src/Command/Asset/HeadAssetCommand.php (hunk 3 above)
Comment (theus77) [reply src/Command/Asset/HeadAssetCommand.php:69]: I can't find this constant?

File: src/Resources/DoctrineMigrations/pdo_pgsql/Version20210412122119.php (hunk 5 above)
Comment (theus77) [reply src/Resources/DoctrineMigrations/pdo_pgsql/Version20210412122119.php:1]: There are maintained. Good point I just forgot

**Review by Davidmattei — DISMISSED** [review]

File: src/Command/Asset/HeadAssetCommand.php (hunk 3 above)
Comment (dameert) [reply src/Command/Asset/HeadAssetCommand.php:65]: Oops its symfony 5 only 👍

**Comment by dameert** [comment]
Before approval EMSCommonBundle should be released as there is a non functional dependency at the moment.

**Review by dameert — APPROVED** [review]

**Pull request merged**